File: functions/src/domain/projectInvitation.bak/repositories/factory/ProjectInvitationRepositoryFactoryImpl.ts
import { ProjectInvitationRepository } from '../projectInvitation.repository';
import { ProjectInvitationRepositoryFactory } from './ProjectInvitationRepositoryFactory';
import { ProjectInvitationRepositoryFactoryContext } from './ProjectInvitationRepositoryFactoryContext';
import { ProjectInvitationRepositoryImpl } from '../../../infrastructure/repositories/projectInvitation.repository.impl';
import { ProjectInvitationDataSource } from '../../../infrastructure/datasources/interfaces/projectInvitation.datasource';
import { ProjectInvitationFirestoreDataSource } from '../../../infrastructure/datasources/firestore/projectInvitation.datasource';
import { getFirestore } from 'firebase-admin/firestore';

/**
 * ProjectInvitationRepository Factory 구현체
 */
export class ProjectInvitationRepositoryFactoryImpl implements ProjectInvitationRepositoryFactory {
  /**
   * ProjectInvitationRepository 인스턴스를 생성합니다.
   * 
   * @param context Factory 컨텍스트
   * @returns ProjectInvitationRepository 인스턴스
   */
  create(context: ProjectInvitationRepositoryFactoryContext): ProjectInvitationRepository {
    // Firestore 데이터소스 생성
    const firestoreDataSource: ProjectInvitationDataSource = new ProjectInvitationFirestoreDataSource(
      getFirestore()
    );

    // Repository 구현체 생성
    return new ProjectInvitationRepositoryImpl(firestoreDataSource);
  }
}

File: data/src/main/java/com/example/data/repository/factory/ProjectInvitationRepositoryFactoryImpl.kt
package com.example.data.repository.factory

import com.example.data.datasource.remote.ProjectInvitationRemoteDataSource
import com.example.data.repository.base.ProjectInvitationRepositoryImpl
import com.example.domain.repository.RepositoryFactory
import com.example.domain.repository.base.ProjectInvitationRepository
import com.example.domain.repository.factory.context.ProjectInvitationRepositoryFactoryContext
import javax.inject.Inject

class ProjectInvitationRepositoryFactoryImpl @Inject constructor(
    private val projectInvitationRemoteDataSource: ProjectInvitationRemoteDataSource,
) : RepositoryFactory<ProjectInvitationRepositoryFactoryContext, ProjectInvitationRepository> {

    override fun create(input: ProjectInvitationRepositoryFactoryContext): ProjectInvitationRepository {
        return ProjectInvitationRepositoryImpl(
            projectInvitationRemoteDataSource = projectInvitationRemoteDataSource,
            factoryContext = input,
        )
    }
} 